server/src/main/java/org/opensearch/indices/recovery/RecoveryTarget.java

```diff
@@ -213,10 +213,13 @@ public void prepareForTranslogOperations(int totalTranslogOps, ActionListener<Vo
         ActionListener.completeWith(listener, () -> {
             state().getIndex().setFileDetailsComplete(); // ops-based recoveries don't send the file details
             state().getTranslog().totalOperations(totalTranslogOps);
+            boolean shouldPerformMigrationBasedTasks = indexShard.shouldSeedRemoteStore() && indexShard.routingEntry().primary();
+            if (shouldPerformMigrationBasedTasks) {
+                indexShard.deleteRemoteStoreContents();
+            }
```

Comment:
lets add comments why the deletion needs to happen before the `indexShard().openEngineAndSkipTranslogRecovery()`.

Reply:
we can remove `indexShard.routingEntry().primary()` check as `shouldSeedRemoteStore` is true only for primary . Instead we can add an assertion for same.  Due to that `shouldPerformMigrationBasedTasks` can be removed .

Reply:
Why not expose this directly in IndexShard?